src/client/components/VoyageDetailsContainer.js

```diff
@@ -1,5 +1,180 @@
-import React from "react";
+import React, { Component } from "react";
+import axios from "axios";
+import VoyageDetails from "./VoyageDetails";
+import CostWidget from "./CostWidget";
+import SuggestedRouteTable from "./SuggestedRouteTable";
 
-export default function VoyageDetailsContainer({ match }) {
-  return <div>This page is about the Voyage ID = {match.params.voyage_id}</div>;
+export default class VoyageDetailsContainer extends Component {
+  state = {
+    voyageId: this.props.match.params.voyage_id,
+    vessel_report: [],
+    suggested_routes: [],
+    vessel_report_id: [],
+    suggested_route_id: [],
+    voyageList: null,
+    selected_route_id: null,
+    selected_route_table: [],
+    departure_time: "",
+    hfocosts: null,
+    lsfocosts: null,
+    vessel_id: null,
+    vesselName: "",
+    departsFrom: "",
+    arrivesAt: "",
+    departure_time: "",
+    vessel_eta: "",
+    date: "",
+    latitude: null,
+    longitude: null,
+    hsfo: null,
+    ulsfo: null,
+    hfo: null,
+    lsfo: null,
+    hfocosts: null,
+    lsfocosts: null,
+    totalcost: null,
+    depart_from_port: null
+    //arrive_at_port: null
+  };
```

Comment:
That is one big state object 😮 

But it's probably fine to do it like that. Excepts that I would expect you to have a full `suggested_route` object in there with the current selected suggested route, rather than just the id of that one. Would also make it easier to get the waypoints.

Reply:
I have the selected route table from there I am getting waypoints